### ISSUE TITLE
Status wallet fails to request permission when using createModernProviderInterface

### ISSUE THREAD

**Issue by chuckbergeron** [blocknative/onboard#465]
I did a bisect and looks like it broken in-between v1.14.0 and v1.15.0. 

When I change Status' interface from:

```
provider && getProviderName(provider) === 'Status'
   ? createModernProviderInterface(provider)
   : null
```

to:
```
interface: {
  name: 'Status',
  connect: () => provider.enable(),
  address: {
    get: () => getAddress(provider)
  },
  network: {
    get: () => getNetwork(provider)
  },
  balance: {
    get: () => getBalance(provider)
  },
}
```

it works flawlessly again.

**Comment by chuckbergeron** [comment]
I won't submit a PR for this as I imagine this isn't the desired way of fixing this issue, but thought I'd let you know!

**Comment by aaronbarnardsound** [comment]
@chuckbergeron Thanks for pointing this out, we will take a look and get a fix out as soon as possible 👍

**Comment by andytudhope** [comment]
Hello @aaronbarnardsound and team! Is this something we can help with? cc @flexsurfer

We don't inject a web3 provider, relying on the .enable() call,which I understand to be best practice and something the rest of the ecosystem has recently moved over to. 

I notice that Status does not show up as a connection option when testing for e.g. Gnosis Safe and would like to make sure it reappears on that list (I see it is in the 1.18 release) asap.

**Comment by aaronbarnardsound** [comment]
Hey @chuckbergeron I did some digging and it looks like the Status module hasn't changed since we first created it (at least not the connection code), so something must have changed on Status' end. I also tried changing to the code you suggested and it was still not working for me on iOS. What are you testing on?

@andytudhope Our module is just checking the provider at `window.ethereum` to see if the `isStatus` property is true as [instructed in your docs](https://status.im/developer_tools/detecting_status.html). If that property is true, then we go ahead and just use the regular provider methods as defined in [eip-1193](https://eips.ethereum.org/EIPS/eip-1193). I am struggling to debug what is going on as the Status browser doesn't seem to allow console logs, but I keep seeing an error with no stack trace and a log that says "Not supported by Status". So any help on this would be appreciated.

Dapps usually define what wallets they would like to display as a connection option, so you will need to reach out to the Gnosis team to ask them to include Status in their wallet selection modal.

**Comment by flexsurfer** [comment]
hi, status doesn't support `.on` methods, so when you are calling `.on` you can see "Not supported by Status" in the logs , is there an URL i could try to reproduce ?

**Comment by aaronbarnardsound** [comment]
Hey @flexsurfer the module will first check to see if the `.on` method is defined before calling it. Does it throw the error if the property is just accessed or when it is called? Is that property defined? You can test on our react demo, live link is [here](https://reactdemo.blocknative.com) and repo [here](https://github.com/blocknative/react-demo).

**Comment by flexsurfer** [comment]
`.on` method is defined in status, but it just does nothing, just prints text in the console. thank you, i will take a look at the demo

**Comment by flexsurfer** [comment]
![image](https://user-images.githubusercontent.com/11790366/106430867-bd4b3400-646c-11eb-8d6e-e5b9aee61f67.png)
this is calls from https://reactdemo.blocknative.com/ in status, and there is no eip-1102 permissions request, https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md , thats why status throws Unauthorized error for `eth_accounts`

**Comment by aaronbarnardsound** [comment]
Ok sweet that should be an easy fix, will take a look and see if we can get this sorted out soon. Thanks!

**Comment by aaronbarnardsound** [comment]
@chuckbergeron This is fixed in version `1.19.1` via PR #491. Thanks @flexsurfer for your help on this one!
